lib/ecto/repo.ex

```diff
@@ -2200,6 +2214,80 @@ defmodule Ecto.Repo do
               | {:error, any}
               | Ecto.Multi.failure()
 
+  @doc """
+  Runs the given function inside a transaction.
+
+  The return value is the same as of the given `fun` which must be
+  `{:ok, result}` or `{:error, reason}`.
+
+  If this function returns `{:ok, result}`, it means the transaction
+  was successfully committed. On the other hand, if it returns `{:error, reason}`,
+  it means the transaction was rolled back.
+
+  If an Elixir exception occurs the transaction will be rolled back
+  and the exception will bubble up from the transaction function.
+  If no exception occurs, the transaction is committed when the
+  function returns. A transaction can be explicitly rolled back
+  by calling `c:rollback/1`, this will immediately leave the function
+  and return the value given to `rollback` as `{:error, value}`.
```

Comment:
i might just be failing at reading comprehension this morning, but this doesn't seem to explicitly detail that the transaction is rolled back if `fun` returns `{:error, reason}` (or really, anything other than `{:ok, result}`)

```suggestion
  If an Elixir exception occurs the transaction will be rolled back
  and the exception will bubble up from the transaction function.
  If no exception occurs, the transaction is committed if the function 
  returns `{:ok, result}`. Returning `{:error, result}` will rollback the transaction
  and this function will return `{:error, result}` as well. 
  A transaction can be explicitly rolled back
  by calling `c:rollback/1`, this will immediately leave the function
  and return the value given to `rollback` as `{:error, value}`.
```